src/ipyaladin/region_converter.py

```diff
@@ -0,0 +1,283 @@
+        RefToLocalRotMatrix
+            The rotation matrix.
+
+        """
+        ca, sa = math.cos(lon), math.sin(lon)
+        cd, sd = math.cos(lat), math.sin(lat)
+        return cls(
+            np.array([[ca * cd, sa * cd, sd], [-sa, ca, 0.0], [-ca * sd, -sa * sd, cd]])
+        )
+
+    def to_global_xyz(self, x: float, y: float, z: float) -> tuple[float, float, float]:
+        """Convert local cartesian coordinates to global cartesian coordinates.
+
+        Parameters
+        ----------
+        x : float
+            The x coordinate.
+        y : float
+            The y coordinate.
+        z : float
+            The z coordinate.
+
+        Returns
+        -------
+        tuple[float, float, float]
+            The x, y, and z coordinates.
+
+        """
+        rotated_matrix = np.sum(
+            self.rotation_matrix * np.array([[x], [y], [z]]), axis=0
+        )
+        return rotated_matrix[0], rotated_matrix[1], rotated_matrix[2]
+
+    def to_global_coo(self, x: float, y: float, z: float) -> tuple[float, float]:
+        """Convert local cartesian coordinates to global spherical coordinates.
+
+        Parameters
+        ----------
+        x : float
+            The x coordinate.
+        y : float
+            The y coordinate.
+        z : float
+            The z coordinate.
+
+        Returns
+        -------
+        tuple[float, float]
+            The longitude and latitude.
+
+        """
+        x, y, z = self.to_global_xyz(x, y, z)
+        # Convert cartesian coordinates to spherical coordinates.
+        r2 = x * x + y * y
+        lat = math.atan2(z, math.sqrt(r2))
+        lon = math.atan2(y, x)
+        if lon < 0:
+            lon += TWICE_PI
+        return lon, lat
+
+
+def rectangle_to_polygon_region(region: RectangleSkyRegion) -> PolygonSkyRegion:
+    """Convert a RectangleSkyRegion to a PolygonSkyRegion.
+
+    Parameters
+    ----------
+    region : RectangleSkyRegion
+        The region to convert.
+
+    Returns
+    -------
+    PolygonSkyRegion
+        The converted region.
+
+    """
+    longitude, latitude = region.center.icrs.ra.rad, region.center.icrs.dec.rad
+    width = region.width.rad
+    height = region.height.rad
+    position_angle = region.angle.rad
+
+    if not 0.0 <= longitude < TWICE_PI:
+        raise ValueError(f"Expected: lon in [0, 2pi[. Actual: {longitude}")
+    if not -math.pi / 2 <= latitude <= math.pi / 2:
+        raise ValueError(f"Expected: lat in [-pi/2, pi/2]. Actual: {latitude}")
+    if not 0.0 < width <= math.pi / 2:
+        raise ValueError(f"Expected: a in ]0, pi/2]. Actual: {width}")
+    if not 0.0 < height <= width:
+        raise ValueError(f"Expected: b in ]0, a]. Actual: {height}")
+    if not 0.0 <= position_angle < math.pi:
+        raise ValueError(f"Expected: pa in [0, pi[. Actual: {position_angle}")
+
+    frame_rotation = RefToLocalRotMatrix.from_center(longitude, latitude)
+
+    sin_lon, cos_lon = math.sin(width), math.cos(width)
```

Comment:
@fxpineau - Do you have the matrix expression of that computation written somewhere? We realise we can directly use it thanks to numpy